### PR TITLE
Feat: Add http2 server support

### PR DESCRIPTION
**Pull request by pulkit4tech**
- [X] Description and context for reviewers: one partner, one stranger
  - Add http2 server support - version2 accept both http1 and http2 request. So all existing inbounds should work as expected. 
  - Add config that can be used to override this 
  - [Test Checklist ](https://docs.google.com/document/d/1mYfH0DmC3Rg61oSV5ulzCVUJsXrHEOuTY4TOfrcTmWM/edit?usp=sharing)
- [X] Entry in CHANGELOG.md

### CODE REVIEW

File: transport/http/inbound.go [hunk 1]
```diff
@@ -124,6 +129,14 @@ func InboundTLSMode(mode yarpctls.Mode) InboundOption {
 	}
 }
 
+// InboundUseHTTP2 returns an InboundOption that sets whether the inbound should
+// accept HTTP/2 requests. If set to false, the inbound will only accept HTTP/1
+func InboundUseHTTP2(useHTTP2 bool) InboundOption {
```
Comment (biosvs) [inline transport/http/inbound.go:134]: Maybe it's personal, but for me it sounds like "http2 is disabled by default". So if we want to enable it by default, it should be "DisableHTTP2", wdyt?

File: transport/http/inbound.go [hunk 2]
```diff
@@ -165,6 +179,9 @@ type Inbound struct {
 
 	tlsConfig *tls.Config
 	tlsMode   yarpctls.Mode
+
+	// default is true
```
Comment (biosvs) [inline transport/http/inbound.go:183]: You see, if it's "disableHttp2", it'd be more straightforward to understand

File: transport/http/inbound.go [hunk 3]
```diff
@@ -221,10 +238,19 @@ func (i *Inbound) start() error {
...
+	}
+	if i.useHTTP2 {
+		h2s := &http2.Server{}
+		server.Handler = h2c.NewHandler(server.Handler, h2s)
+		err := http2.ConfigureServer(server, h2s)
+		if err != nil {
+			return fmt.Errorf("failed to configure HTTP/2 server: %w", err)
+		}
+	}
+	i.server = intnet.NewHTTPServer(server)
```
Comment (biosvs) [inline transport/http/inbound.go:253]: Let's write some real integration tests: with running server on some port, making real http2-request, etc.

**Review by biosvs — CHANGES_REQUESTED** [review]

File: transport/http/inbound.go (hunk 1 above)
Comment (pulkit4tech) [reply transport/http/inbound.go:134]: yep, it make sense, but as we are introducing new support (http2) I thought options should be explicitly mentioning this usage. I can change this as I don't have strong opinion on this :)

**Pull request merged**